packages/waku/src/router/define-router.tsx

```diff
@@ -635,113 +632,100 @@ export function unstable_defineRouter(fns: {
     const cachedElementsForBuild = new Map<SlotId, Promise<ReactNode>>();
     const serializedCachedElements = new Map<SlotId, string>();
     const getCachedElement = (id: SlotId) => cachedElementsForBuild.get(id);
-    const setCachedElement =
-      (req?: Request) => (id: SlotId, element: ReactNode) => {
-        const cached = cachedElementsForBuild.get(id);
-        if (cached) {
-          return cached;
-        }
-        const fn = async () => {
-          const teedStream = renderRsc({ [id]: element }).then((rscStream) =>
-            rscStream.tee(),
-          );
-          const stream1 = teedStream.then(([s1]) => s1);
-          const stream2 = teedStream.then(([, s2]) => s2);
-          const copied = stream1.then(
-            (rscStream) =>
-              parseRsc(rscStream).then(
-                (parsed) => parsed[id],
-              ) as Promise<ReactNode>,
-          );
-          cachedElementsForBuild.set(id, copied);
-          serializedCachedElements.set(id, await streamToBase64(await stream2));
-          return copied;
-        };
-        if (req) {
-          return withRequest(req, fn);
-        } else {
-          return fn();
-        }
-      };
+    const setCachedElement = async (id: SlotId, element: ReactNode) => {
+      const cached = cachedElementsForBuild.get(id);
+      if (cached) {
+        return cached;
+      }
+      const teedStream = renderRsc({ [id]: element }).then((rscStream) =>
+        rscStream.tee(),
+      );
+      const stream1 = teedStream.then(([s1]) => s1);
+      const stream2 = teedStream.then(([, s2]) => s2);
+      const copied = stream1.then(
+        (rscStream) =>
+          parseRsc(rscStream).then(
+            (parsed) => parsed[id],
+          ) as Promise<ReactNode>,
+      );
+      cachedElementsForBuild.set(id, copied);
+      serializedCachedElements.set(id, await streamToBase64(await stream2));
+      return copied;
+    };
+
+    // hard-coded concurrency limit
+    const { runTask, waitForTasks } = createTaskRunner(500);
```

Comment:
Related to https://github.com/wakujs/waku/pull/1776#issuecomment-3539571812, I'm guessing that the reason why throttling needs to be inside router level (and not each `emitFile`) is because new static elements doesn't exactly correspond to single static file. Is that the case?

If so, we have no way to make "partial mode" to save static rendering.

Reply:
https://github.com/wakujs/waku/pull/1776#issuecomment-3539707569

> Is that the case?

Because of #1744, we now eagerly renders some components. Yeah, you got the point.

Reply:
> If so, we have no way to make "partial mode" to save static rendering.

...good catch.

Reply:
> But, speaking of partial mode, I think directly reusing RSC stream of old build is inherently sketchy and I'm not sure we should support such concept.

I agree.